sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs

```diff
@@ -231,7 +231,8 @@ await _sessionReceiver.AbandonMessageAsync(message, propertiesToModify, cancella
         /// </summary>
         public virtual void ReleaseSession() =>
             // manager will be null if instance created using the public constructor which is exposed for testing purposes
-            _manager?.CancelSession();
+            // This will be awaited when closing the receiver.
+            _ = _manager?.CancelSessionAsync();
```

Comment:
Should we consider attaching a `ContinueWith` scoped to exceptions to capture any problems in the logs?

Reply:
We will end up awaiting this task when the session is closed, which would happen either when switching to a new session or when closing the processor. If for some reason the application crashes and we never await the task, we would still log non-TCE errors within the [RenewSessionLock](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs#L378) task. I think using ContinueWith would be equivalent to the current behavior as it would still only execute once the task completes.